src/libraries/System.Data.OleDb/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1256,27 +1254,12 @@ internal static bool RemoveStringQuotes(string? quotePrefix, string? quoteSuffix
 
         internal static IntPtr IntPtrOffset(IntPtr pbase, int offset)
         {
-            if (4 == ADP.PtrSize)
-            {
-                return (IntPtr)checked(pbase.ToInt32() + offset);
-            }
-            Debug.Assert(8 == ADP.PtrSize, "8 != IntPtr.Size");
-            return (IntPtr)checked(pbase.ToInt64() + offset);
+            return (nint)pbase + offset;
         }
 
-        internal static int IntPtrToInt32(IntPtr value)
+        internal static int IntPtrToInt32(nint value)
         {
-            if (4 == ADP.PtrSize)
-            {
-                return (int)value;
-            }
-            else
-            {
-                long lval = (long)value;
-                lval = Math.Min((long)int.MaxValue, lval);
-                lval = Math.Max((long)int.MinValue, lval);
-                return (int)lval;
-            }
+            return (int)value;
```

Comment:
Note that this is a behavior change from the previous code.

If you are going to change this, I think it should be `return checked((int)value);` so that the overflows are not silently ignored.

Reply:
Also, this is only used in 3 places. It may be better to delete this method and just inline this into the 3 places where it is used.

Reply:
OK, I thought the workaround:
```cs
long lval = (long)value;
lval = Math.Min((long)int.MaxValue, lval);
lval = Math.Max((long)int.MinValue, lval);
```
 is actually for avoiding the overflow, no? And now as it will not throw when overflowing this change should be safe

Reply:
Yes, the original code is trying to avoid the overflow by clipping the value to int.MinValue/int.MaxValue range.

`(int)value;` is not going to clip the value to int.MinValue/int.MaxValue range. It is just going to return the low 32 bits. That's where the behavior change is. This should either keep the original clipping behavior; or it should be changed to throw exception on overflow.

Reply:
Right, then I think its better to keep the old behavior, thanks!